### PR TITLE
chore: change MTU only if not Azure CNI on Azure Stack

### PR DESCRIPTION
**Pull request by jadarsie**
**Reason for Change**:

Azure CNI is not using encap, no need to reduce MTU on Azure Stack

**Issue Fixed**:

Related #1346

**Requirements**:

- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

### CODE REVIEW

File: parts/k8s/cloud-init/artifacts/cse_customcloud.sh [hunk 1]
```diff
@@ -80,14 +80,15 @@ configureK8sCustomCloud() {
...
     echo $(cat "${AZURE_JSON_PATH}" | jq '.tenantId = "adfs"') >${AZURE_JSON_PATH}
   fi
+  set -x
 
-  # Decrease eth0 MTU to mitigate Azure Stack's NRP issue
-  echo "iface eth0 inet dhcp" | sudo tee -a /etc/network/interfaces
-  echo "    post-up /sbin/ifconfig eth0 mtu 1350" | sudo tee -a /etc/network/interfaces
-
-  ifconfig eth0 mtu 1350
+    {{- if not IsAzureCNI}}
```
Comment (jackfrancis) [inline parts/k8s/cloud-init/artifacts/cse_customcloud.sh:85]: nit: unnecessary indentation

(the pattern we've been following thus far is not to indent actual static chars [e.g., the shell script in this case], so you can put the `{{` and the echo statements on a common indentation indentation alignment at the beginning of the line)

**Comment by jackfrancis** [comment]
@jadarsie let me know when you've tested this successfully in an Azure Stack environment, and I'll merge

**Comment by jadarsie** [comment]
> @jadarsie let me know when you've tested this successfully in an Azure Stack environment, and I'll merge

**Comment by jadarsie** [comment]
> @jadarsie let me know when you've tested this successfully in an Azure Stack environment, and I'll merge

this is good to go

**Pull request merged**